File: go/regex/regex_test.go
package regex

import "testing"

func TestRegex(test *testing.T) {

	// Basic match, beginning of string
	if match("foo","foobar") != 1 { test.Fail() }

	// Basic match, middle of string
	if match("oba","foobar") != 1 { test.Fail() }

	// Basic match, no match 
	if match("obo","foobar") != 0 { test.Fail() }

	// Match with start qualifier
	if match("^fo","foobar") != 1 { test.Fail() }

	// Match with start qualifier in body
	if match("^bar","foobar") != 0 { test.Fail() }

	// Match with end qualifier 
	if match("bar$","foobar") != 1 { test.Fail() }

	// Match with end qualifier in body
	if match("foo$","foobar") != 0 { test.Fail() }

	// Match with optional qualifier
	if match("fo*b","foobar") != 1 { test.Fail() }

	// Match with optional qualifier 2
	if match("fooa*b","foobar") != 1 { test.Fail() }

	// Match with optional qualifier 3
	if match("a*foo","foobar") != 1 { test.Fail() }
}

File: python/regex_test.py
import unittest

from regex import match

class regex_test (unittest.TestCase):

    def testRegex(self):

        # Basic match, beginning of string
        self.assertEqual(1,match("foo","foobar"))

        # Basic match, middle of string
        self.assertEqual(1,match("oba","foobar"))

        # Basic match, no match 
	self.assertEqual(0,match("obo","foobar"))

        # Match with start qualifier
        self.assertEqual(1,match("^fo","foobar"))

        # Match with start qualifier in body
        self.assertEqual(0,match("^bar","foobar"))

        # Match with end qualifier 
        self.assertEqual(1,match("bar$","foobar"))

        # Match with end qualifier in body
        self.assertEqual(0,match("foo$","foobar"))

        # Match with optional qualifier
        self.assertEqual(1,match("fo*b","foobar"))

        # Match with optional qualifier 2
        self.assertEqual(1,match("fooa*b","foobar"))

        # Match with optional qualifier 3
        self.assertEqual(1,match("a*foo","foobar"))


if __name__ == '__main__':
    unittest.main()
